### PR TITLE
Test to assert guest user has permission denied to create or upload files

### LINKED ISSUE #36712: Add acceptance test to assert that the guest users have permission denied to create or upload files
- [ ] - Add acceptance test to assert that the guest users have permission denied to create or upload files

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add accepatance test to assert that the guest user has permission denied to create or upload files

## Related Issue

- Related to : #36712 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/lib/FilesPage.php [hunk 1]
```diff
@@ -43,6 +43,7 @@ class FilesPage extends FilesPageBasic {
 	protected $deleteAllSelectedBtnXpath = ".//*[@id='app-content-files']//*[@class='delete-selected']";
 	protected $homePageIconXpath = "//div[@class='breadcrumb']//img[@alt='Home']";
 	protected $folderBreadCrumbXpath = "//div[@class='breadcrumb']//a[contains(@href,'%s')]";
+	protected $deniedPermissionToUploadCreate = ".notCreatable.notPublic";
```
Comment (HariBhandari07) [inline tests/acceptance/features/lib/FilesPage.php]: we usually append `Xpath` at the last. IMO its better if the xpath variable is noun.

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 2]
```diff
@@ -2244,4 +2244,14 @@ public function publicLinkWithLastShareTokenShouldBeListedAsShareReceiverViaOnTh
...
 	}
+
+	/**
+	 * @Then the user should not have permission to upload or create files
+	 *
+	 * @return void
+	 */
+	public function deniedPermissionToUploadAndCreateFiles() {
+		$isVisible = $this->filesPage->permissionDeniedToUploadCreateFiles();
+		Assert::assertTrue($isVisible);
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: Also check the displayed message is `You don’t have permission to upload or create files here`

File: tests/acceptance/features/lib/FilesPage.php (hunk 1 above)
Comment (individual-it) [reply tests/acceptance/features/lib/FilesPage.php]: this is not an xpath but a CSS property

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebUIFilesContext.php]: fixed

File: tests/acceptance/features/lib/FilesPage.php (hunk 1 above)
Comment (HariBhandari07) [reply tests/acceptance/features/lib/FilesPage.php]: Suggestion `uploadCreatePermissionDeniedMessageSelector`

Comment (jasson99) [reply tests/acceptance/features/lib/FilesPage.php]: fixed

File: tests/acceptance/features/lib/FilesPage.php [hunk 3]
```diff
@@ -413,4 +414,17 @@ public function isSharedIndicatorPresent($fileName, $session) {
 		$fileRow = $this->findFileRowByName($fileName, $session);
 		return $fileRow->isSharedIndicatorPresent();
 	}
+
+	/**
+	 * checks whether the permission is denied to create and upload files
+	 *
+	 * @return bool
+	 */
+	public function permissionDeniedToUploadCreateFiles() {
```
Comment (HariBhandari07) [inline tests/acceptance/features/lib/FilesPage.php]: `getUploadCreatePermissionDeniedMessage`

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 4]
```diff
@@ -2244,4 +2244,15 @@ public function publicLinkWithLastShareTokenShouldBeListedAsShareReceiverViaOnTh
 		$shareTreeItem = $sharingDialog->getShareTreeItem("public link", $token, $item);
 		Assert::assertTrue($shareTreeItem->isVisible());
 	}
+
+	/**
+	 * @Then the user should not have permission to upload or create files
+	 *
+	 * @return void
+	 */
+	public function deniedPermissionToUploadAndCreateFiles() {
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: `assertUploadCreatePermissionIsDenied`

File: tests/acceptance/features/lib/FilesPage.php [hunk 5]
```diff
@@ -413,4 +414,17 @@ public function isSharedIndicatorPresent($fileName, $session) {
 		$fileRow = $this->findFileRowByName($fileName, $session);
 		return $fileRow->isSharedIndicatorPresent();
 	}
+
+	/**
+	 * checks whether the permission is denied to create and upload files
```
Comment (HariBhandari07) [inline tests/acceptance/features/lib/FilesPage.php]: `gets create-upload permission message`

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php (hunk 4 above)
Comment (HariBhandari07) [reply tests/acceptance/features/bootstrap/WebUIFilesContext.php]: Also check `+` is not displayed in the webUI

File: tests/acceptance/features/lib/FilesPage.php [hunk 6]
```diff
@@ -413,4 +414,17 @@ public function isSharedIndicatorPresent($fileName, $session) {
...
+	 *
+	 * @return bool
```
Comment (individual-it) [inline tests/acceptance/features/lib/FilesPage.php]: you don't return bool values anywhere

**Review by individual-it — DISMISSED** [review]

File: tests/acceptance/features/lib/FilesPage.php (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/features/lib/FilesPage.php]: fixed

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebUIFilesContext.php]: fixed

File: tests/acceptance/features/lib/FilesPage.php (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/features/lib/FilesPage.php]: fixed

File: tests/acceptance/features/lib/FilesPage.php (hunk 6 above)
Comment (jasson99) [reply tests/acceptance/features/lib/FilesPage.php]: fixed

**Comment by HariBhandari07** [comment]
LGTM

**Review by haribhandari07 — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Pull request merged**